### PR TITLE
validate ephemeral models when hub is launched in non-production environment

### PR DESCRIPTION
**Pull request by habdelra**

### CODE REVIEW

File: packages/hub/main.js [hunk 1]
```diff
@@ -53,6 +50,33 @@ async function wireItUp(projectDir, encryptionKeys, dataSources, opts = {}) {
   return container;
 }
 
+async function startIndexing(environment, container) {
+  // some datasources are dependent upon a sync at boot for index of pristine system
+  await container.lookup('hub:indexers').update();
+
+  if (environment !== 'production') {
+    let ephemeralStorage = await container.lookup(`plugin-services:${require.resolve('@cardstack/ephemeral/service')}`);
```
Comment (ef4) [inline packages/hub/main.js]: I think this will throw if you don't happen to have ephemeral installed?

File: packages/hub/main.js [hunk 2]
```diff
@@ -53,6 +50,33 @@ async function wireItUp(projectDir, encryptionKeys, dataSources, opts = {}) {
...
+    let searchers = await container.lookup(`hub:searchers`);
+    let controllingBranch = await container.lookup(`hub:controlling-branch`);
+    let models = await (await container.lookup('config:initial-models'))();
+
+    await ephemeralStorage.validateModels(models, async (type, id) => {
+      let result;
+      try {
+        result = await searchers.get(Session.INTERNAL_PRIVILEGED, controllingBranch.name, type, id);
```
Comment (ef4) [inline packages/hub/main.js]: There is a `searchers.getFromControllingBranch()` so you don't need to lookup `hub:controlling-branch` yourself.

File: packages/hub/main.js (hunk 1 above)
Comment (habdelra) [reply packages/hub/main.js]: good point

File: packages/hub/main.js (hunk 2 above)
Comment (habdelra) [reply packages/hub/main.js]: nice. thanks!

**Comment by habdelra** [comment]
@ef4 ok, I've resolved the issue that you raised. hopefully this is ready for merging now.

**Pull request merged**